### DISCUSSION TITLE
OpenTelemetry.Instrumentation.SqlClient Library and metrics

### DISCUSSION THREAD

**Discussion by noelmcgrath** [open-telemetry/opentelemetry-dotnet#5196] [Q&A]
New to Open Telemetry so forgive question if the answer is obvious.
The SqlClient only has traces and no metrics. 

My question is why is there no metrics for around database calls?

For example you have a service that makes http calls and reads\saves to database.

For the http calls we can use OpenTelemetry.Instrumentation.Http and get metric [http.client.request.duration](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-metrics.md#metric-httpclientrequestduration) to find out how long request took.
Should the same type of metrics be available for observing how long database calls are taken?

I believe this can be achieved by looking at the traces, I may be wrong as have not dived into tracing yet, but is there also a place for sql metrics to allow u to view where your service time is been spent and identify bottlenecks etc. over a period of time.

I would think it was not overlooked to not include metrics, so there must be a reason that I am not getting and would appreciate some guidance.
Thanks

**Comment by cijothomas** [comment]
The instrumentation libraries generally follow the OTel semantic conventions. As of today the conventions for database are here : https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/database/database-metrics.md.
It is definitely possible for one to enhance the existing sql instrumentaion to add metrics support!
Here is the tracking issue : https://github.com/open-telemetry/opentelemetry-dotnet/issues/2497 and you may consider contributing to it.
